File: src/BetterLyrics.Sdk/Interfaces/Plugins/ILyricsSource.cs
using BetterLyrics.Sdk.Models.Lyrics;

namespace BetterLyrics.Sdk.Interfaces.Plugins;

public interface ILyricsSource
{
    /// <summary>
    /// </summary>
    /// <param name="title"></param>
    /// <param name="artist"></param>
    /// <param name="album"></param>
    /// <param name="duration"></param>
    /// <param name="token"></param>
    /// <exception cref="OperationCanceledException"></exception>
    /// <returns></returns>
    Task<LyricsSearchResult> GetLyricsAsync(string title, string artist, string album, double duration,
        CancellationToken token);
}

File: src/BetterLyrics.KMP/shared/src/commonMain/kotlin/com/jayfunc/betterlyrics/sdk/interfaces/plugins/ILyricsSource.kt
package com.jayfunc.betterlyrics.sdk.interfaces.plugins

import com.jayfunc.betterlyrics.sdk.models.lyrics.LyricsSearchResult

interface ILyricsSource {
    /**
     * @param title
     * @param artist
     * @param album
     * @param duration
     * @throws CancellationException if the coroutine is cancelled during the operation
     * @return LyricsSearchResult
     */
    suspend fun getLyricsAsync(
        title: String,
        artist: String,
        album: String,
        duration: Double
    ): LyricsSearchResult
}
